sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -1515,6 +1516,58 @@ async def __Delete(self, path, request_params, req_headers, **kwargs):
             request_data=None,
             **kwargs
         )
+    
+    async def Batch(self, collection_link, batch_operations, options=None, **kwargs):
+        """Executes the given operations in transactional batch.
+
+        :param str collection_link: The link to the collection
+        :param list batch_operations: The batch of operations for the batch request.
+        :param dict options: The request options for the request.
+
+        :return:
+            The result of the batch operation.
+        :rtype:
+            dict
+
+        """
+        if options is None:
+            options = {}
+
+        path = base.GetPathFromLink(collection_link, "docs")
+        collection_id = base.GetResourceIdOrFullNameFromLink(collection_link)
+
+        batch_operations = base._format_batch_operations(batch_operations)
+
+        result, self.last_response_headers = await self._Batch(
+            batch_operations,
+            path,
+            collection_id,
+            options,
+            **kwargs)
+
+        final_responses = []
+        is_error = False
+        for i in range(len(result)):
+            status_code = result[i].get("statusCode")
+            if status_code >= 400:
+                is_error = True
+                final_responses.append(models.BatchOperationError(operation=batch_operations[i],
+                                                                  operation_response=result[i],
+                                                                  error=Constants.ERROR_TRANSLATIONS.get(status_code)))
+            else:
+                final_responses.append(models.BatchOperationResponse(operation=batch_operations[i],
```

Comment:
there is one more check I see in Java SDK side is to make sure for successful situation, the returned results would match the operation size, else throw internal server error
https://github.com/Azure/azure-sdk-for-java/blob/07f29bfd599460dfa7071f4f1f3de41257db28df/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BatchResponseParser.java#L75